### ISSUE TITLE
Add crash report functionality

### ISSUE THREAD

**Issue by lwouis** [lwouis/alt-tab-macos#132]
I know that there are established libraries to handle crash reports in the mac OSS. This ticket is about discussing adding such a library so that alt-tab share a crash report on crash, or even non-crashing errors. How we collect this data (opt-in, opt-out, out much data, privacy, etc) is very open. Keep in mind this project is OSS and I do want to preserve that spirit by being very transparent and non-invasive.

A first pass would be to list the available libraries that provide this type of functionality, and listing also what they do.

**Comment by lwouis** [comment]
Bartender uses [FRFeedbackReporter](https://github.com/tcurdt/feedbackreporter) to report feedback

**Comment by lwouis** [comment]
I investigated FRFeedbackReporter. It doesn't let us decide how the data is sent. It's assuming an HTTP server with a specific endpoint. I checked DevMate also, and this one is just huge and full of features.

I think I'll roll my own UI as there is very little to be done. I thought about how the feedback will be shared given that this is OSS. My idea so far is to make a HTTP POST call to github to open a ticket. I'll create an account, generate a token for it, and embed the token in the app, so that the app submits ticket on behalf of that (bot) account.

Here is the UI I have so far, compared to other apps:

![image](https://user-images.githubusercontent.com/106195/73839449-5eeb7580-4859-11ea-9f01-d0e560f9487d.png)

**Comment by lwouis** [comment]
For crash reporting, I investigated some more, and it seems like a very high complexity process. Swift is built on top of many layers, and crashes/exceptions/signals can happen in any of the C, C++, objc, mach layer. I think using an establish library here is the only realistic way to go.

I found 2 interesting popular libraries:

- [plcrashreporter](https://github.com/microsoft/plcrashreporter)
- [KSCrash](https://github.com/kstenerud/KSCrash)
